chainercv/links/model/feature_extraction_predictor.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+import warnings
+
+import chainer
+from chainer import cuda
+
+from chainercv.transforms import center_crop
+from chainercv.transforms import resize
+from chainercv.transforms import scale
+from chainercv.transforms import ten_crop
+
+
+class FeatureExtractionPredictor(chainer.Chain):
+
+    """Wrapper that adds a prediction method to a feature extraction model.
+
+    The :meth:`predict` takes three steps to make a prediction.
+
+    1. Preprocess input images
+    2. Forward the preprocessed images to the network
+    3. Average features in the case when more than one crops are extracted.
+
+    Example:
+
+        >>> from chainercv.links import VGG16
+        >>> from chainercv.links import FeatureExtractionPredictor
+        >>> base_model = VGG16()
+        >>> model = FeatureExtractionPredictor(base_model)
+        >>> prob = model.predict([img])
+        # Predicting multiple features
+        >>> model.extractor.feature_names = ['conv5_3', 'fc7']
+        >>> conv5_3, fc7 = model.predict([img])
+
+    When :obj:`self.crop == 'center'`, :meth:`predict` extracts features from
+    the center crop of the input images.
+    When :obj:`self.crop == '10'`, :meth:`predict` extracts features from
+    patches that are ten-cropped from the input images.
+
+    When extracting more than one crops from an image, the output of
+    :meth:`predict` returns the average of the features computed from the
+    crops.
+
+    Args:
+        extractor: A feature extraction model. This is a callable chain
+            that takes a batch of images and returns a variable or a
+            tuple of variables.
+        crop_size (int or tuple): The height and the width of an image after
+            cropping in preprocessing.
+            If this is an integer, the image is cropped to
+            :math:`(crop_size, crop_size)`.
+        scale_size (int or tuple): If :obj:`scale_size` is an integer, during
+            preprocessing, an image is resized so that the length of the
+            shorter edge is equal to :obj:`scale_size`. If this is a tuple
+            :obj:`(height, width)`, the image is resized to
+            :math:`(height, width)`.
```

Comment:
How about supporting `do not scale`? Some user may want this feature if all images are larger than `crop_size`.